src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -1376,27 +1379,32 @@ def __repr__(self):
                                                          id=self.daemon_id)
 
     def to_json(self):
-        out = {
-            'hostname': self.hostname,
-            'container_id': self.container_id,
-            'container_image_id': self.container_image_id,
-            'container_image_name': self.container_image_name,
-            'daemon_id': self.daemon_id,
-            'daemon_type': self.daemon_type,
-            'version': self.version,
-            'status': self.status,
-            'status_desc': self.status_desc,
-        }
+        out = OrderedDict()
+        out['daemon_type'] = self.daemon_type
+        out['daemon_id'] = self.daemon_id
+        out['hostname'] = self.hostname
+        out['container_id'] = self.container_id
+        out['container_image_id'] = self.container_image_id
+        out['container_image_name'] = self.container_image_name
+        out['version'] = self.version
+        out['status'] = self.status
+        out['status_desc'] = self.status_desc
+
         for k in ['last_refresh', 'created', 'started', 'last_deployed',
                   'last_configured']:
             if getattr(self, k):
                 out[k] = getattr(self, k).strftime(DATEFMT)
-        return {k: v for (k, v) in out.items() if v is not None}
+
+        empty = [k for k, v in out.items() if v is None]
+        for e in empty:
+            del out[e]
```

Comment:
That's confusing, why not just use regular for?
```
for k,v in out.items():
  if not v:
     del out[k]
```

Reply:
No, as it will modify dictionary size during iteration.

Reply:
> No, as it will modify dictionary size during iteration.

It should not be a problem, however, if you unsure, you can could use only keys() and construct a list from it:
```
for k in list(out.keys()):
  if not out[k]:
     del out[k]
```

Reply:
yeah, that's what I'm doing here.